quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java

```diff
@@ -17,13 +17,145 @@
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_REALM = new OptionBuilder<>("storage-realm", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_CLIENT = new OptionBuilder<>("storage-client", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_CLIENT_SCOPE = new OptionBuilder<>("storage-client-scope", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_GROUP = new OptionBuilder<>("storage-group", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_ROLE = new OptionBuilder<>("storage-role", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER = new OptionBuilder<>("storage-user", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_DEPLOYMENT_STATE = new OptionBuilder<>("storage-deployment-state", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_AUTH_SESSION = new OptionBuilder<>("storage-auth-session", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER_SESSION = new OptionBuilder<>("storage-user-session", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_LOGIN_FAILURE = new OptionBuilder<>("storage-login-failure", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_AUTHORIZATION_PERSISTER = new OptionBuilder<>("storage-authorization-persister", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER_SESSION_PERSISTER = new OptionBuilder<>("storage-user-session-persister", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
```

Comment:
I thought we discussed this one. We need to use here the `DisabledUserSessionPersisterProvider`.

Reply:
Still - can this go away since #12712 disables the JPA implementation?

Reply:
Exactly, once we have that we can remove this option.

Do not worry too much about options now because they are going to change once we review not only this but other providers that should be marked as env dependent. The options are internal.